addons/sourcemod/scripting/nd_building_fixes.sp

```diff
@@ -4,138 +4,213 @@
 
 //Version is auto-filled by the travis builder
-public Plugin myinfo = 
+public Plugin myinfo =
 {
-	name 		= "[ND] Buildings Fixes",
-	author 		= "stickz",
-    description	= "Prevent building walls inside relays/repeaters",
-    version 	= "dummy",
-	url 		= "https://github.com/stickz/Redstone/"
+    name        = "[ND] Buildings Fixes",
+    author      = "stickz, databomb",
+    description = "Prevent building walls inside relays/repeaters",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
 }
 
+char g_sPowerStructErrorMessage[2][64] = {"CANNOT BUILD TOO NEAR WIRELESS REPEATER.", "CANNOT BUILD TOO NEAR RELAY TOWER."};
+
 /* Auto-Updater Support */
 #define UPDATE_URL  "https://github.com/stickz/Redstone/raw/build/updater/nd_building_fixes/nd_building_fixes.txt"
 #include "updater/standard.sp"
 
 #define DISTANCE_INSIDE_RELAY 60
 
-public void OnPluginStart() {
-	AddUpdaterLibrary(); //auto-updater	
+// Check for nd_structure_intercept being available
+bool g_bStructureDetourAvailable = true;
+public void OnAllPluginsLoaded()
+{
+    g_bStructureDetourAvailable = LibraryExists("nd_structure_intercept");
+    if (!g_bStructureDetourAvailable)
+    {
+        LogError("Problem was found and automatically reverting to ND_OnStructureCreated. Check gamedata.");
+    }
+}
+
+public void OnPluginStart()
+{
+    AddUpdaterLibrary(); //auto-updater
 }
 
+public Action ND_OnCommanderBuildStructure(int client, ND_Structures &structure, float position[3])
+{
+    if (!ND_RoundStarted())
+        return Plugin_Continue;
+
+    if (structure == ND_Relay_Tower || structure == ND_Wireless_Repeater)
+    {
+        // Get the team the relay tower or wireless repeater belongs to
+        int relayTeam = GetClientTeam(client);
+
+        if (IsPowerTooCloseToWall(relayTeam, position))
+        {
+            UTIL_Commander_FailureText(client, "CANNOT BUILD TOO NEAR WALL.");
+            return Plugin_Stop;
+        }
+    }
+    else if (structure == ND_Wall)
+    {
+        // Get the team the wall belongs to
+        int wallTeam = GetClientTeam(client);
+
+        if (IsWallTooCloseToPower(wallTeam, position))
+        {
+            UTIL_Commander_FailureText(client, g_sPowerStructErrorMessage[wallTeam-2]);
+            return Plugin_Stop;
+        }
+    }
+
+    return Plugin_Continue;
+}
 public void ND_OnStructureCreated(int entity, const char[] classname)
-{	
-	if (!ND_RoundStarted())
-		return;
-	
-	if (ND_IsStructRelay(classname)) {
-		CreateTimer(0.1, CheckRelay, entity);		
-	}
-	else if (StrEqual(classname, STRUCT_WALL, true)) {
-		CreateTimer(0.1, CheckWall, entity);
-	}		
+{
+    if (!g_bStructureDetourAvailable && ND_RoundStarted())
+    {
+        int entref = EntIndexToEntRef(entity);
+        if (ND_IsStructRelay(classname))
+        {
+            CreateTimer(0.1, Timer_CheckRelay, entref, TIMER_FLAG_NO_MAPCHANGE);
+        }
+        else if (!strcmp(classname, STRUCT_WALL))
```

Comment:
We should use `StrEqual` here instead. This is confusing, more intensive and not required.

Reply:
StrEqual adds additional overhead and is not recommended.

Reply:
Changed to StrEqual for readability